### PR TITLE
add r-base

### PR DESCRIPTION
**Pull request by ocefpaf**
@pelson I appreciate if you could take a look. Not sure if this is all we need to unlock the creation of `R` feedstocks.

### CODE REVIEW

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -478,12 +478,12 @@ def update_matrix(forge_config, new_matrix):
...
     """
-    Turns a matrix with environment variables and pakages into a matrix of
+    Turns a matrix with environment variables and packages into a matrix of
     just environment variables. The package variables are prefixed with CONDA,
-    and special cases such as Python and Numpy are handled.
+    and special cases such as Python, NumPy, and R are handled.
 
     """
-    special_conda_vars = {'python': 'CONDA_PY', 'numpy': 'CONDA_NPY'}
+    special_conda_vars = {'python': 'CONDA_PY', 'numpy': 'CONDA_NPY', 'r-base': 'CONDA_R'}
```
Comment (jakirkham) [inline conda_smithy/configure_feedstock.py]: It's interesting that we don't do this with `perl`, yet we still get a matrix with it. So am unclear if this is needed at this time. Though may need to look at this more closely.

**Comment by jakirkham** [comment]
Have you tried using this to run `conda smithy init` on the `r-boot` recipe? I just tried this and am not seeing `CONDA_R` show up in the CI matrices. 😕 Not to mention am still seeing the original issue of [`r-base` unsatisfiable on `osx-64`]( https://travis-ci.org/conda-forge/staged-recipes/builds/198760898#L300-L301 ). 😒

File: conda_smithy/configure_feedstock.py (hunk 1 above)
Comment (johanneskoester) [inline conda_smithy/configure_feedstock.py]: I have also some doubts about removing the dots in the version (line 498). Is that really needed? It won't work for R because at least the R channel pins to 3.3.2 and 332 is pretty ambiguous.

**Comment by johanneskoester** [comment]
I would like to help, but for me conda-smithy fails like that:

```
File "/home/johannes/scms/conda-forge/conda-smithy/conda_smithy/configure_feedstock.py", line 517, in copytree
    copytree(s, d, ignore, root_dst=root_dst)
  File "/home/johannes/scms/conda-forge/conda-smithy/conda_smithy/configure_feedstock.py", line 517, in copytree
    copytree(s, d, ignore, root_dst=root_dst)
  File "/home/johannes/scms/conda-forge/conda-smithy/conda_smithy/configure_feedstock.py", line 517, in copytree
    copytree(s, d, ignore, root_dst=root_dst)
  File "/home/johannes/scms/conda-forge/conda-smithy/conda_smithy/configure_feedstock.py", line 517, in copytree
    copytree(s, d, ignore, root_dst=root_dst)
  File "/home/johannes/scms/conda-forge/conda-smithy/conda_smithy/configure_feedstock.py", line 517, in copytree
    copytree(s, d, ignore, root_dst=root_dst)
  File "/home/johannes/scms/conda-forge/conda-smithy/conda_smithy/configure_feedstock.py", line 516, in copytree
    os.makedirs(d)
  File "/home/johannes/.local/opt/miniconda/lib/python3.5/os.py", line 241, in makedirs
    mkdir(name, mode)
OSError: [Errno 36] File name too long: './r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock/recipe/r-boot-feedstock'
```

**Comment by jakirkham** [comment]
Thanks for taking a look @johanneskoester. That's a fun error. Some things to try:

1. Please update `conda-smithy` to latest (`2.0.1` currently).
2. Please make sure you are using `conda` 4.2 (`conda-build-all` and `conda-smithy` are not 4.3 compatible ATM).
3. Please move/copy `r-boot` out of any existing repo (e.g. `staged-recipes`) if you haven't already.
4. Please simply run `conda smithy init r-boot`.

If you still have issues, please let us know version information and more details about what is being run and how.

**Comment by johanneskoester** [comment]
Thanks. Downgrading conda to 4.2 works. So, I checked it and it creates an appropriate matrix entry in the variable `forge_config`. The matrix is correctly written to `.travis.yml`. However, it is not written to `circle.yml`. This is because the corresponding block in the template is missing, see [here](https://github.com/conda-forge/conda-smithy/blob/master/conda_smithy/templates/circle.yml.tmpl). I would say this is a bug independent of this PR. It also won't write a CONDA_PY etc. entry.

In addition, I checked again and we really should not remove the dots in the version numbers. This is also  not necessary for Python and Numpy and so on.

**Comment by johanneskoester** [comment]
@ocefpaf  I would appreciate if this would be quickly solved and merged. Moving R to conda-forge is really pressing now for Bioconda. If you don't have the time, please let me know, and I'll create a separate PR with the necessary fixes.

**Comment by ocefpaf** [comment]
@johanneskoester let me see what I can do today and then I will pass it to you tomorrow.

**Comment by jakirkham** [comment]
Could you share some info on how this is configured and run @johanneskoester? Right now, I'm still seeing this error. Note this is the same error referenced [above]( https://github.com/conda-forge/conda-smithy/pull/453#issuecomment-279249157 ).

```bash
$ conda smithy init r-boot/
No appveyor token. Create a token at https://ci.appveyor.com/api-token and
Put one in ~/.conda-smithy/appveyor.token
No anaconda token. Create a token via
  anaconda auth --create --name conda-smithy --scopes "repos conda api"
and put it in ~/.conda-smithy/anaconda.token
Fetching package metadata .............
Fetching package metadata .............
Fetching package metadata .............
Fetching package metadata ...............
Package missing in current osx-64 channels: 
  - r-base
```

Though I am now seeing `CONDA_R` show up in the matrices. Not sure why I didn't see it before, but am now seeing it.

> However, it is not written to `circle.yml`.

This gets written into `ci_support/run_docker_build.sh`, which it appears to be doing correctly. It goes into the script directly as CircleCI doesn't have matrix support (unless they have added this very recently). This works by using [this template]( https://github.com/conda-forge/conda-smithy/blob/master/conda_smithy/templates/run_docker_build_matrix.tmpl ).

File: conda_smithy/configure_feedstock.py (hunk 1 above)
Comment (jakirkham) [inline conda_smithy/configure_feedstock.py]: I think `conda-build` doesn't need this any more. Though am unclear on whether we or `conda-build-all` is relying on these in particular format.

Comment (jakirkham) [inline conda_smithy/configure_feedstock.py]: Just tried removing `r-base`/`CONDA_R` from here and it still shows up in the matrix.

File: conda_smithy/configure_feedstock.py [hunk 2]
```diff
@@ -563,7 +563,7 @@ def compute_build_matrix(meta, existing_matrix=None, channel_sources=tuple()):
     index = conda.api.get_index(channel_urls=channel_sources,
                                 platform=meta_config(meta).subdir)
     mtx = special_case_version_matrix(meta, index)
-    mtx = list(filter_cases(mtx, ['python >=2.7,<3|>=3.5', 'numpy >=1.11']))
+    mtx = list(filter_cases(mtx, ['python >=2.7,<3|>=3.5', 'numpy >=1.11', 'r-base >=3.3.2']))
```
Comment (jakirkham) [inline conda_smithy/configure_feedstock.py]: Actually, if I remove this, it still re-renders correctly. Strange.

**Comment by johanneskoester** [comment]
@jakirkham ok then the matrix is written correctly. Hence, the only remaining issue is to get rid of the dot removal @ocefpaf.

@jakirkham I can't think of a reason why your conda does not see r-base. It is clearly available in conda-forge, also for osx: https://anaconda.org/conda-forge/r-base/files

Does conda install r-base work for you?

**Comment by jakirkham** [comment]
Please see my edit in the comment above. Basically I think the issue is it doesn't see it for `win-32`, but is misreporting this. Not sure why it is wrong about the platform or why it says this at all.

File: conda_smithy/configure_feedstock.py (hunk 1 above)
Comment (jakirkham) [inline conda_smithy/configure_feedstock.py]: I think the reason we can get away with not including this is `conda-build-all` handles adding `CONDA_R` for us.

**Comment by jakirkham** [comment]
Looking a little closer, I think the error message is actually coming from `conda` itself. `conda` still uses `subdir` in some cases where it should prefer the platform provided via the API and so it fails the `win-32` re-render, but reports that it was using my OS hence `osx-64`. This is why we see the same error at staged-recipes. Would expect on Linux one might see `linux-64` (assuming it is 64-bit). Do you see such a message in your logs, @johanneskoester?

cc @kalefranz @msarahan

**Comment by johanneskoester** [comment]
Nope, for me, it ran without error. I used conda 4.2.16 and conda-build 2.1.1.

**Comment by jakirkham** [comment]
Also as another test. I tried running `conda smithy init` using the current release and it does create the matrix. However, the re-rendering in all of these cases is incomplete. For instance, if I go into the feedstock directory created and run `git status`, then `git` complains it is not a repo. Also no `README.md` is created. Do you see any of these issues?

**Comment by johanneskoester** [comment]
No, for me the rendering is complete and exits without error:

```
$ conda-smithy init r-boot
Fetching package metadata .............
Fetching package metadata .............
Fetching package metadata .............
Fetching package metadata ...............
Fetching package metadata ...............
Initialized empty Git repository in /tmp/test/r-boot-feedstock/.git/
warning: LF will be replaced by CRLF in recipe/bld.bat.
The file will have its original line endings in your working directory.
[master (root-commit) 8e26f9b] Initial feedstock commit with conda-smithy 0.post.dev628.
warning: LF will be replaced by CRLF in recipe/bld.bat.
The file will have its original line endings in your working directory.
 12 files changed, 344 insertions(+)
 create mode 100644 .gitattributes
 create mode 100644 .gitignore
 create mode 100644 .travis.yml
 create mode 100644 LICENSE
 create mode 100644 README.md
 create mode 100755 ci_support/checkout_merge_commit.sh
 create mode 100755 ci_support/run_docker_build.sh
 create mode 100644 circle.yml
 create mode 100644 conda-forge.yml
 create mode 100644 recipe/bld.bat
 create mode 100644 recipe/build.sh
 create mode 100644 recipe/meta.yaml

Repository created, please edit conda-forge.yml to configure the upload channels
and afterwards call 'conda smithy register-github'
```

**Comment by jakirkham** [comment]
> I used conda 4.2.16 and conda-build 2.1.1.

Tried those versions and am still seeing this issue. 😕 Can you please provide `conda list -n root`?

**Comment by johanneskoester** [comment]
If I understand that correct, I would say that you run into a bug with conda that only occurs on osx when somehow faking that it is on windows, right? Could we handle this is a separate issue and still merge this, as the new functionality itself works ok?

File: conda_smithy/configure_feedstock.py (hunk 1 above)
Comment (ocefpaf) [inline conda_smithy/configure_feedstock.py]: I need this here for my local workflow. Please do not remove it without a major version version bump!

File: conda_smithy/configure_feedstock.py [hunk 3]
```diff
@@ -495,7 +494,8 @@ def prepare_matrix_for_env_vars(matrix):
                 name, value = item
                 if name in special_conda_vars:
                     name = special_conda_vars[name]
-                    value = str(value).replace('.', '')
+                    if name.upper() != 'CONDA_R':
```
Comment (ocefpaf) [inline conda_smithy/configure_feedstock.py]: @johanneskoester I am reluctant to change this to everything as you suggested, even though I am sure it would work maybe we should leave that to another PR.

**Comment by ocefpaf** [comment]
Merging this and I will leave the mystery of the `win-32` issue to you @johanneskoester. I will be out for a few days.

File: conda_smithy/configure_feedstock.py (hunk 3 above)
Comment (johanneskoester) [inline conda_smithy/configure_feedstock.py]: Ok.

**Comment by jakirkham** [comment]
> If I understand that correct, I would say that you run into a bug with conda that only occurs on osx when somehow faking that it is on windows, right? Could we handle this is a separate issue and still merge this, as the new functionality itself works ok?

What I'm trying to say is I can reproduce a bug locally that occurs at staged-recipes with or without this change. The goal of this change was to fix that problem, but it doesn't. So we still will be unable to convert `r-boot` or other R recipes at staged-recipes.

**Comment by johanneskoester** [comment]
So, we should maybe compare the conda versions. What conda version do you have locally?

**Comment by jakirkham** [comment]
I installed the same versions you have and reproduced the issue. Hence the question of what else do you have in `root`? Could you please provide a list?

**Comment by johanneskoester** [comment]
Note that I use conda-smithy from this branch though.

```
_nb_ext_conf              0.3.0                    py35_0  
alabaster                 0.7.9                    py35_0  
anaconda-client           1.5.3                    py35_0  
arrow                     0.10.0                    <pip>
babel                     2.3.4                    py35_0  
bcftools                  1.3.1                         1    bioconda
beautifulsoup4            4.5.3                    py35_0  
beautifulsoup4            4.5.1                     <pip>
binaryornot               0.4.0                     <pip>
boost                     1.60.0                   py35_0  
bs4                       0.0.1                     <pip>
bwa                       0.7.15                        0    bioconda
bzip2                     1.0.6                         3  
cairo                     1.12.18                       6  
cairocffi                 0.7.2                    py35_0    local
cairosvg                  2.0.0rc6                 py35_0    conda-forge
cffi                      1.8.3                    py35_0    file:///home/johannes/scms/snakemake/tests/test_conda/.snakemake/conda-archive/6291eeeb63ef1987c7d7fa4494eb56d6
chardet                   2.3.0                     <pip>
chardet                   2.3.0                    py35_0  
clever-toolkit            2.0rc3          py35_boost1.60_1    bioconda
click                     6.7                       <pip>
cloog                     0.18.0                        0  
clyent                    1.2.2                    py35_0  
conda                     4.2.16                   py35_0  
conda-build               2.1.1                    py35_0  
conda-build-all           1.0.0                    py35_0    conda-forge
conda-env                 2.6.0                         0  
conda-smithy              1.6.0                    py35_0    conda-forge
conda-verify              2.0.0                    py35_0  
cookiecutter              1.5.0                     <pip>
cryptography              1.5.2                    py35_0  
cssselect                 0.9.2                    py35_0  
curl                      7.49.0                        1  
cycler                    0.10.0                   py35_0  
dbus                      1.10.10                       0  
decorator                 4.0.10                   py35_0  
decorator                 4.0.10                    <pip>
docutils                  0.12                     py35_2  
dropbox                   5.2.1                    py35_0    bioconda
entrypoints               0.2.2                    py35_0  
expat                     2.1.0                         0  
filechunkio               1.6                      py35_0    file:///home/johannes/scms/snakemake/tests/test_conda/.snakemake/conda-archive/6291eeeb63ef1987c7d7fa4494eb56d6
filelock                  2.0.6                    py35_0  
fontconfig                2.11.1                        6  
freebayes                 1.1.0                    py35_0    bioconda
freetype                  2.5.5                         1  
ftputil                   3.2                      py35_0    bioconda
future                    0.16.0                    <pip>
gitdb                     0.6.4                    py35_0    conda-forge
gitpython                 2.0.8                    py35_1    conda-forge
glib                      2.43.0                        1  
gmp                       6.1.0                         0  
gsl                       1.16                          0    conda-forge
gst-plugins-base          1.8.0                         0  
gstreamer                 1.8.0                         0  
harfbuzz                  0.9.39                        1  
icu                       54.1                          0  
idna                      2.1                      py35_0  
imagesize                 0.7.1                    py35_0  
ipykernel                 4.5.1                    py35_0  
ipython                   5.1.0                    py35_0  
ipython_genutils          0.1.0                    py35_0  
ipywidgets                5.2.2                    py35_0  
isl                       0.12.2                        0  
jbig                      2.1                           0  
jinja2                    2.8                      py35_1  
jinja2-time               0.2.0                     <pip>
jpeg                      8d                            2  
jsonschema                2.5.1                     <pip>
jsonschema                2.5.1                    py35_0  
jupyter                   1.0.0                    py35_3  
jupyter_client            4.4.0                    py35_0  
jupyter_console           5.0.0                    py35_0  
jupyter_core              4.2.0                    py35_0  
libffi                    3.2.1                         0  
libgcc                    5.2.0                         0  
libgd                     2.2.3                         1    bioconda
libgfortran               3.0.0                         1  
libpng                    1.6.22                        0  
libsodium                 1.0.10                        0  
libtiff                   4.0.6                         2  
libxcb                    1.12                          1  
libxml2                   2.9.2                         0  
libxslt                   1.1.28                        0  
lxml                      3.6.4                    py35_0  
markupsafe                0.23                     py35_2  
matplotlib                1.5.3               np111py35_0  
Merfishtools              1.0.0                     <pip>
mistune                   0.7.3                    py35_0  
mkl                       11.3.3                        0  
mock                      2.0.0                    py35_0  
mpc                       1.0.3                         0  
mpfr                      3.1.5                         0  
nb_anacondacloud          1.2.0                    py35_0  
nb_conda                  2.0.0                    py35_0  
nb_conda_kernels          2.0.0                    py35_0  
nbconvert                 4.2.0                    py35_0  
nbformat                  4.1.0                    py35_0  
nbpresent                 3.0.2                    py35_0  
ncurses                   5.9                           8  
networkx                  1.11                      <pip>
nose                      1.3.7                    py35_1  
notebook                  4.2.3                    py35_0  
numpy                     1.11.2                   py35_0  
openjdk                   8.17.0.3                      1  
openssl                   1.0.2h                        1  
pandas                    0.19.2              np111py35_1  
pango                     1.39.0                        1  
paramiko                  2.0.2                    py35_0  
patchelf                  0.9                           0  
path.py                   8.2.1                    py35_0  
pbr                       1.10.0                   py35_0  
pcre                      8.39                          0  
pexpect                   4.0.1                    py35_0  
pickleshare               0.7.4                    py35_0  
pillow                    3.4.2                    py35_0  
pip                       8.1.2                    py35_0  
pixman                    0.32.6                        0  
pkginfo                   1.3.2                    py35_0  
pockets                   0.3.1                     <pip>
poyo                      0.4.0                     <pip>
prompt_toolkit            1.0.9                    py35_0  
prosic                    2.0.0                 gsl1.16_0    local
ptyprocess                0.5.1                    py35_0  
py                        1.4.32                   py35_0  
pyaml                     15.8.2                   py35_0    bioconda
pyasn1                    0.1.9                    py35_0  
pycosat                   0.6.1                    py35_1  
pycparser                 2.14                     py35_1  
pycrypto                  2.6.1                    py35_4  
pygithub                  1.29                     py35_0    conda-forge
pygments                  2.1.3                    py35_0  
pyopenssl                 16.2.0                   py35_0  
pyparsing                 2.1.4                    py35_0  
pyqt                      5.6.0                    py35_0  
pysftp                    0.2.8                    py35_0    bioconda
pytest                    3.0.5                    py35_0  
python                    3.5.2                         0  
python-dateutil           2.5.3                    py35_0  
pytz                      2016.7                   py35_0  
pyyaml                    3.11                     py35_4  
pyzmq                     16.0.2                   py35_0  
qt                        5.6.0                         1  
qtconsole                 4.2.1                    py35_1  
r                         3.3.1                  r3.3.1_0    r
r-base                    3.3.1                         3    r
r-boot                    1.3_18                 r3.3.1_0    r
r-class                   7.3_14                 r3.3.1_0    r
r-cluster                 2.0.4                  r3.3.1_0    r
r-codetools               0.2_14                 r3.3.1_0    r
r-foreign                 0.8_66                 r3.3.1_0    r
r-kernsmooth              2.23_15                r3.3.1_0    r
r-lattice                 0.20_33                r3.3.1_0    r
r-mass                    7.3_45                 r3.3.1_0    r
r-matrix                  1.2_6                  r3.3.1_0    r
r-mgcv                    1.8_12                 r3.3.1_0    r
r-nlme                    3.1_128                r3.3.1_0    r
r-nnet                    7.3_12                 r3.3.1_0    r
r-recommended             3.3.1                  r3.3.1_0    r
r-rpart                   4.1_10                 r3.3.1_0    r
r-spatial                 7.3_11                 r3.3.1_0    r
r-survival                2.39_4                 r3.3.1_0    r
readline                  6.2                           2  
requests                  2.12.4                   py35_0  
rpy2                      2.8.1              py35r3.3.1_1    r
ruamel.yaml               0.12.13                  py35_0    bioconda
ruamel_yaml               0.11.14                  py35_0  
samtools                  1.3.1                         4    bioconda
scipy                     0.18.1              np111py35_0  
seaborn                   0.7.1                    py35_0  
setuptools                23.0.0                   py35_0  
simplegeneric             0.8.1                    py35_1  
sip                       4.18                     py35_0  
six                       1.10.0                   py35_0  
smmap                     0.9.0                    py35_0    conda-forge
snakemake                 3.10.1                   py35_1    bioconda
snowballstemmer           1.2.1                    py35_0  
sphinx                    1.5                      py35_0  
sphinx_rtd_theme          0.1.9                    py35_0  
sphinxcontrib-napoleon    0.6.0                     <pip>
sqlite                    3.13.0                        0  
terminado                 0.6                      py35_0  
tinycss                   0.4                      py35_0    conda-forge
tk                        8.5.18                        0  
tornado                   4.4.2                    py35_0  
traitlets                 4.3.1                    py35_0  
typing                    3.5.2.2                  py35_0    bioconda
urllib3                   1.12                     py35_0    bioconda
wcwidth                   0.1.7                    py35_0  
wheel                     0.29.0                   py35_0  
whichcraft                0.4.0                     <pip>
widgetsnbextension        1.2.6                    py35_0  
wrapt                     1.10.8                   py35_0  
xz                        5.2.2                         0  
yaml                      0.1.6                         0  
zeromq                    4.1.5                         0  
zlib                      1.2.8                         3
```

**Comment by jakirkham** [comment]
Thanks @johanneskoester. I'll try to compare these and get back to you.

FWIW was also able to reproduce this in the `linux-anvil` docker container. So I don't think this is an OS X isolated issue. There is something funky between the `conda` tools in play and am just unclear where the problem is coming from ATM.

**Comment by jakirkham** [comment]
So the listing says you have `conda-smithy` version `1.6.0` in `root`? How did you install this development version of `conda-smithy`?

**Comment by johanneskoester** [comment]
`pip install -e .`

**Comment by jakirkham** [comment]
Also I see you have `conda-build-all` version `1.0.0`. I thought that wouldn't work for R. 😕

**Comment by johanneskoester** [comment]
I also have the git version of conda-build-all, installed via `pip install -e .`. Conda just does not see that.

**Comment by jakirkham** [comment]
So what version is it actually?

**Comment by johanneskoester** [comment]
It was the one from the r-base pull request. However, I just updated it to the latest release (1.0.1-py35_1), and it works fine. No errors when generating the feedstock, exactly the same output as I posted above.

**Comment by jakirkham** [comment]
Latest release should be 1.0.2.

**Comment by jakirkham** [comment]
Interesting it appears not to be getting `conda-build-all` from the conda-forge channel, which channel is it getting it from.

**Comment by johanneskoester** [comment]
You are right. I think it came from default, because I have a higher priority for default (to be able to use the R channel together with conda-forge). Now I have conda-build-all 1.0.2 and conda 4.2.13 from conda-forge.
I can still create the feedstock without any error.

**Comment by jakirkham** [comment]
Sorry I don't know what to say. Unfortunately I don't have more time to debug this, but I do want to provide you a way for with R that will work. Am proposing we make a slight hack to the `r-boot` recipe that should allow it to generate a feedstock. Please see PR ( https://github.com/conda-forge/staged-recipes/pull/2446 ).
